### PR TITLE
Handle throwables too

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

File: src/Runner/Runner.php [hunk 1]
```diff
@@ -168,14 +168,9 @@ private function fixFile(\SplFileInfo $file, LintingResultInterface $lintingResu
...
-            $this->dispatchEvent(
-                FixerFileProcessedEvent::NAME,
-                FixerFileProcessedEvent::create()->setStatus(FixerFileProcessedEvent::STATUS_EXCEPTION)
-            );
-
-            $this->errorsManager->report(new Error(Error::TYPE_EXCEPTION, $name));
-
-            return;
+            return $this->processException($name);
+        } catch (\Throwable $e) {
```
Comment (keradus) [inline src/Runner/Runner.php]: it's sad to use class that doesn't exists on some env like php5

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: that's fine? php doesn't care if the class doesn't exist

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: catching both exception and then throwable in this order is common place these days

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: Example: https://github.com/laravel/framework/blob/v5.2.29/src/Illuminate/Queue/SyncQueue.php#L33-L45.

Comment (keradus) [inline src/Runner/Runner.php]: I know PHP won't crash on it, but still it's ugly. Want to play with PHP7 features? use polyfill

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: there is no polyfill for this that's not "shit"

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: this way here is by far the best way to handle this, and that's why everyone does it this way

Comment (keradus) [inline src/Runner/Runner.php]: https://github.com/symfony/polyfill/blob/master/src/Php70/Resources/stubs/Error.php

Trowable = Error + Exception

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: > Trowable = Error + Exception

Not necessarily true. Something can implement Throwable without extending Error or Exception.

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: This way here is future proof.

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: Symfony itself doesn't even catch it's own polyfil errors, it catches throwable.

**Comment by GrahamCampbell** [comment]
Is this good to merge @keradus?

**Comment by keradus** [comment]
I don't like the solution but still see the value of it.

For that I'm waiting for any other contributors to give opinion ;)

File: src/Runner/Runner.php [hunk 2]
```diff
@@ -232,6 +227,21 @@ private function fixFile(\SplFileInfo $file, LintingResultInterface $lintingResu
     }
 
     /**
+     * Process an exception that occurred.
+     *
+     * @param string $name
+     */
+    protected function processException($name)
```
Comment (gharlan) [inline src/Runner/Runner.php]: why not `private` like other methods in this class?

**Comment by gharlan** [comment]
👍 (apart from my comment above)

File: src/Runner/Runner.php (hunk 2 above)
Comment (GrahamCampbell) [inline src/Runner/Runner.php]: sorry, just a habit i have from laravel

Comment (GrahamCampbell) [inline src/Runner/Runner.php]: i'll get that changed

**Comment by GrahamCampbell** [comment]
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1882

**Pull request closed without merging**